rust/src/wallet.rs

```diff
@@ -182,6 +195,48 @@ impl Wallet {
     pub fn get_node_id(&self) -> PublicKey {
         self.lightning.channel_manager.get_our_node_id()
     }
+
+    pub fn get_script_status(&self, script: Script, txid: Txid) -> Result<ScriptStatus> {
+        let script_status = self
+            .lightning
+            .wallet
+            .get_tx_status_for_script(script, txid)
+            .map_err(|_| anyhow!("Could not get tx status for script"))?;
+        Ok(script_status)
+    }
+
+    pub fn send_to_address(&self, send_to: Address, amount: u64) -> Result<Txid> {
+        let wallet = self.lightning.wallet.get_wallet();
+
+        let estimated_fee_rate = self
+            .lightning
+            .wallet
+            .estimate_fee(ConfirmationTarget::Normal)
+            .map_err(|_| anyhow!("Failed to estimate fee"))?;
+
+        let (mut psbt, _) = {
+            let mut builder = wallet.build_tx();
+            builder
+                .add_recipient(send_to.script_pubkey(), amount)
+                .enable_rbf()
+                .do_not_spend_change()
+                .fee_rate(FeeRate::from_sat_per_vb(
+                    f32::from_u32(estimated_fee_rate).unwrap_or(1.0),
+                ));
+            builder.finish()?
+        };
+
+        if !wallet.sign(&mut psbt, SignOptions::default())? {
+            bail!("Failed to sign psbt");
+        }
+
+        let tx = psbt.extract_tx();
+
+        // TODO: This swallows the result internally - if we fail to broadcast we'll never know :/
+        self.lightning.wallet.broadcast_transaction(&tx);
```

Comment:
👍 Nice that this exists. Can we fix the problem in the TODO easily? 

The alternative is to build an Electrum client here. Calling `transaction_broadcast` on it does return a `Result`.

Reply:
I'm just fixing this TODO with a PR against `bdk-ldk`